Jetpack Button Block: Ensure block enqueues core Button block styles as a dependency

andrewserong opened the pull request:
As tracked in https://github.com/Automattic/jetpack/issues/19464 the Jetpack Button block depends on styles from the core Button block (via the class name `.wp-block-button__link`. In an FSE theme like TT1-blocks, no styles are provided to override the button block styling, so it becomes apparent that when no core Button block is inserted into a template part, then the Jetpack button block appears unstyled.

To address this, we can ensure that the core Button block styles are always enqueued when a Jetpack Button block is rendered on the server.

<!--- Provide a general summary of your changes in the Title above -->
<!-- Would you like this feature to be tested by Beta testers?
Please add testing instructions to to-test.md in a new commit as part of your PR. -->

Possible fix for #19464 and related to https://github.com/Automattic/jetpack/issues/19504

#### Changes proposed in this Pull Request:
<!--- Explain what functional changes your PR includes -->
* Always ensure the core Button Block styles are enqueued when rendering a Jetpack Button block
* Provide a default for the Button border (`inherit`), to avoid the browser default button border from being added

#### Jetpack product discussion
<!-- If you're an Automattician, include a shortlink to the p2 discussion with Jetpack Product here. -->
<!-- Make sure any changes to existing products have been discussed and agreed upon -->

#### Does this pull request change what data or activity we track or use?
<!--- If so, please add the "[Status] Needs Privacy Updates" label and explain what changes there are. -->
<!--- Check existing Jetpack support documents for a preview of the information we need. -->
No

| Before | After |
| --- | --- |
| ![image](https://user-images.githubusercontent.com/14988353/115515657-b5ab6680-a2c8-11eb-8bfc-ed6d65b5d162.png) | ![image](https://user-images.githubusercontent.com/14988353/115516188-379b8f80-a2c9-11eb-8617-30828536b333.png) |

Ensuring no regressions on a non-FSE site, checking that outline style and custom colours still works:

![image](https://user-images.githubusercontent.com/14988353/115516099-2488bf80-a2c9-11eb-914d-9c6b5515fc5e.png)


#### Testing instructions:
<!-- If you were reviewing this PR, how would you like the instructions to be presented? -->
<!-- Please include detailed testing steps, explaining how to test your change. -->
<!-- Bear in mind that context you working on is not obvious for everyone.  -->
<!-- Adding "simple" configuration steps will help reviewers to get to your PR as quickly as possible. -->
<!-- "Before / After" screenshots can also be very helpful when the change is visual. -->

* In an FSE site with the TT1-blocks theme active, insert a block that uses the Jetpack Button block (e.g. Mailchimp or Contact Form) into a template part like the footer template part.
* Save the template.
* View the front end of the site — you should see the default styling of the Button block is applied to the Jetpack Button block
* On a non-FSE theme and site, using a Varia theme like Barnsbury, insert one of the blocks that uses the Jetpack Button block, and ensure that on the front end it still renders as you'd expect using the theme's styles for the button

andrewserong commented on projects/plugins/jetpack/extensions/blocks/button/button.php:
```diff
@@ -39,10 +39,15 @@ function register_block() {
 function render_block( $attributes, $content ) {
 	$save_in_post_content = get_attribute( $attributes, 'saveInPostContent' );
 
-	if ( Blocks::is_amp_request() ) {
```
Note: this AMP is removed here. Previously we only loaded the block's style if it's an AMP request because the only styles we needed to load were targeted for AMP. We now have a style rule that should be applied when not in an AMP request, so I think the conditional is no longer needed.

stacimc reviewed (commented):
This tested mostly well for me, and looks like it will allow us to handle some of the related issues (Mailchimp for example) more easily!

I tested with Contact Form, Mailchimp, Donations, Payments, and Premium Content. I tried to test with Pay with Paypal but it is currently pretty broken/being worked on. A couple of things I noticed:

Button alignment does not appear to be working correctly. In FSE for example, I inserted a Contact Form and tried to make the button 50% width and center-aligned. The width applies but the alignment does not:

<img width="1040" alt="Screen Shot 2021-04-21 at 10 21 43 AM" src="https://user-images.githubusercontent.com/63313398/115595140-6e52b500-a28b-11eb-893f-0cc584378fd5.png">

I also noticed that the Premium Content buttons are stacked on top of each other, and width/alignment do not function correctly:

<img width="529" alt="Screen Shot 2021-04-21 at 10 21 50 AM" src="https://user-images.githubusercontent.com/63313398/115595831-4021a500-a28c-11eb-81a3-89140fb7c500.png">

Do you think it would be worth also enqueuing the `Buttons` block styles as well? I believe that would at least fix the vertical stacking on the Premium Content buttons, although I'm inclined to say most of the above should be separate issues from this PR -- it looks like the styles for alignment classes aren't all loaded, and some of the Premium Content issues appear to be block-specific.

andrewserong commented:
Thanks for testing this out @stacimc! I'll have a play around with enqueuing the Buttons block styles, too. I think if the Jetpack Button block was designed to depend on the styling from those blocks, it's fair to enqueue them as well.

andrewserong commented:
@stacimc thanks again for the thorough testing! I've updated this to also enqueue the `core/buttons` styles, which appears to fix the stacking buttons issue with the premium content block:

![image](https://user-images.githubusercontent.com/14988353/115667022-a855b100-a388-11eb-8f4c-24bb65254eed.png)

For the issue with `center` alignment, it looks like this has been fixed in Gutenberg and will land in GB 10.5 (I double-checked on wpcom with the Gutenberg Edge sticker activated, which is now running 10.5.0 RC1, and it looks like it's working). The PR that fixed it is: https://github.com/WordPress/gutenberg/pull/30525

I've also added a CSS rule for right alignment for the Jetpack Button block within the editor (I noticed right alignment wasn't displaying correctly for e.g. the button block in the Mailchimp block).

There are a bunch of other little issues with button alignment, but like you mention, these are probably better tackled as separate issues/PRs for the individual blocks affected. After this PR lands, I'm happy to work through some of those to fix them up a bit. It might step slightly outside of scope for FSE strictly-speaking, but it'd be good to get some of them sorted.

Do you think this PR is looking okay now for a first pass, or do you think it needs tweaking / addressing other issues too?

stacimc reviewed (approved):
This looks great and will have a big impact 🥳 Tests well for me and I totally agree about the rest being handled in separate PRs. Thanks!

jeherve reviewed (approved):
This looks good to me. 🚢

andrewserong commented:
r224816-wpcom

oskosk commented on projects/plugins/jetpack/extensions/blocks/button/view.scss:
```diff
@@ -1,3 +1,7 @@
 }
+
+.wp-block-jetpack-button button {
+	border: inherit;
```
This change seems to be related to https://github.com/Automattic/themes/issues/5776#issuecomment-1080932897
